solr/core/src/test/org/apache/solr/cli/TestSolrCLIRunExample.java

```diff
@@ -558,32 +555,34 @@ public void testInteractiveSolrCloudExample() throws Exception {
 
   @Test
   public void testFailExecuteScript() throws Exception {
-    File solrHomeDir = new File(ExternalPaths.SERVER_HOME);
-    if (!solrHomeDir.isDirectory())
-      fail(solrHomeDir.getAbsolutePath() + " not found and is required to run this test!");
+    Path solrHomeDir = Path.of(ExternalPaths.SERVER_HOME);
+    if (!Files.isDirectory(solrHomeDir))
+      fail(solrHomeDir.toAbsolutePath() + " not found and is required to run this test!");
 
-    Path tmpDir = createTempDir();
-    File solrExampleDir = tmpDir.toFile();
-    File solrServerDir = solrHomeDir.getParentFile();
+    Path solrExampleDir = createTempDir();
+    Path solrServerDir = solrHomeDir.getParent();
 
     // need a port to start the example server on
     int bindPort = -1;
     try (ServerSocket socket = new ServerSocket(0)) {
       bindPort = socket.getLocalPort();
     }
 
-    File toExecute = new File(tmpDir.toString(), "failExecuteScript");
-    assertTrue(
-        "Should have been able to create file '" + toExecute.getAbsolutePath() + "' ",
-        toExecute.createNewFile());
+    Path toExecute = Path.of(solrExampleDir.toString(), "failExecuteScript");
+    try {
+      Files.createFile(toExecute);
+    } catch (IOException e) {
+      throw new IOException(
+          "Should have been able to create file '" + toExecute.toAbsolutePath() + "' ");
+    }
 
     String[] toolArgs =
         new String[] {
           "-e", "techproducts",
-          "--server-dir", solrServerDir.getAbsolutePath(),
-          "--example-dir", solrExampleDir.getAbsolutePath(),
+          "--server-dir", solrServerDir.toAbsolutePath().toString(),
```

Comment:
Shame that this requires all these `.toString`..   Is there a smarter way of handling this toolArgs?   `var toolArgs`?  And then internally we call .toString on anything passed in?????